content/docs/atom-guide/dev-guide/shaders/azsl/_index.md

```diff
@@ -1,20 +1,468 @@
+  
+Classes has no constructors/destructors and as such, do not accept default member initializers.
+
+### Matrices
+AZSL follows the HLSL convention to use column major matrices by default.  
+  
+However, in the O3DE Runtime, the default convention is row major matrices.  
+  
+When declaring a matrix, the order can be specified using the row_major or column_major keywords.  
+  
+    ShaderResourceGroupSemantic MatrixExample
+    {
+        FrequencyId = 1;
+    };
+      
+    ShaderResourceGroup MatrixExampleSRG : MatrixExample
+    {
+                     float2x3 defaultMatrix;
+        row_major    float2x3 rowMajorMatrix;
+        column_major float2x3 colMajorMatrix;
+    };
+  
+### Arrays And Unbounded Arrays
+From the point of view of defining arrays, the rules and limitations are exactly the same as HLSL. Examples:  
+  
+    ShaderResourceGroup PerObject : BindingPerObject
+    {
+        float m_arrayOfFloats[64]; // GOOD. Arrays of fundamental types must be of constant size.
+     
+        struct SomeStruct
+        {
+            int m_integer;
+            bool m_boolean;
+        }
+        static const int MyArraySize = 32;
+        SomeStruct m_arrayOfStructs[MyArraySize]; // GOOD. Arrays of packable types (structs & fundamentals) require constant size definition.
+     
+        int m_unboundedArrayOfIntegers[]; // ERROR. Unbounded arrays are not supported for fundamental(packables, in general) data types.
+     
+        Texture2D<float4> m_textures[MyArraySize]; // GOOD. Contant size array of Resource View types is ok.
+         
+        ConstantBuffer<SomeStruct> m_unboundedArrayOfStructs[]; // GOOD. Unbounded array of Resource View type is ok. BUT, there are some restrictions.
+    }
+  
+To learn more about what are the limitations when declaring unbounded arrays, go here: [Binding Rules For Unbounded Arrays.](unbounded-arrays)  
+  
+### <a name="partial"></a> Partial ShaderResourceGroup Definitions.
+Before talking about `partial ShaderResourceGroup`, it's important to talk about [ShaderResourceGroupSemantic](srg-semantics), which is always required when declaring a `ShaderResourceGroup`.  
+Go to this section to read about [ShaderResourceGroupSemantic](srg-semantics).  
+  
+The `partial` keyword is useful to define parts of an SRG in different blocks. For example, across different files. This is a similar concept as partial classes in C#.  
+There are no limits into how many `partial` blocks can be used to define an SRG.  
+There are only three requirements:  
+1. When using `partial` to define an SRG block, all other SRG definition blocks, for the same SRG, must start with the `partial` keyword.
+2. For a given SRG, at least one `partial` SRG block must define the Shader Resource Group Semantic.
+3. For a given SRG, if the Shader Resource Group Semantic is defined more than once across different `partial` blocks, they must refer to the same `ShaderResourceGroupSemantic`.  
```

Comment:
- Added the first sentence to clarify the code used to define partial SRGs.
```suggestion
Define a partial SRG by using `partial ShaderResourceGroup`. The `partial` keyword is useful to define parts of an SRG in different blocks (for example, across different files). This is a similar concept as partial classes in C#.  There are no limits on how many `partial` blocks define an SRG.

There are three requirements for defining partial SRGs:
1. When using `partial` to define an SRG block, all other SRG definition blocks within the same SRG must also start with the `partial` keyword.  
2. For a given SRG, at least one partial SRG block must define the SRG Semantic.  
3. For a given SRG, if the SRG Semantic is defined more than once across different partial blocks, then they must refer to the same `ShaderResourceGroupSemantic` object.   
```

Reply:
Fixed.